This patch is to add the register dumps in the cpu info proc. Currently added to dump SEGMENT CONTROL Registers

diff --git a/arch/mips/kernel/proc.c b/arch/mips/kernel/proc.c
--- a/arch/mips/kernel/proc.c
+++ b/arch/mips/kernel/proc.c
@@ -128,6 +128,9 @@
 	seq_printf(m, fmt, 'D', vced_count);
 	seq_printf(m, fmt, 'I', vcei_count);
 	seq_printf(m, "\n");
+#ifdef CONFIG_PROC_LTQ_DEBUG
+	seq_printf(m,"segment control registers 0:%.8x 1:%.8x 2:%.8x\n",read_c0_segctl0(),read_c0_segctl1(),read_c0_segctl2());
+#endif
 
 	return 0;
 }

diff --git a/fs/proc/Kconfig b/fs/proc/Kconfig
--- a/fs/proc/Kconfig
+++ b/fs/proc/Kconfig
@@ -72,3 +72,7 @@
 	default n
 	depends on EXPERT
 	bool "Strip non-essential /proc functionality to reduce code size"
+
+config PROC_LTQ_DEBUG
+	default n
+	bool "LTQ Debug information in the /proc system"

